sys/linux/net_80211.txt

```diff
@@ -55,3 +64,192 @@ define JOIN_IBSS_BG_SCAN	0x1
 define JOIN_IBSS_BG_NO_SCAN	0x2
 
 join_ibss_modes = JOIN_IBSS_NO_SCAN, JOIN_IBSS_BG_SCAN, JOIN_IBSS_BG_NO_SCAN
+
+################################################################################
+#  Common fields and enums
+################################################################################
+
+# As defined in drivers/net/wireless/mac80211_hwsim.c
+ieee80211_channel_freq_mhz = 2412, 2417, 2422, 2427, 2432, 2437, 2442, 2447, 2452, 2457, 2462, 2467, 2472, 2484, 5180, 5200, 5220, 5240, 5260, 5280, 5300, 5320, 5500, 5520, 5540, 5560, 5580, 5600, 5620, 5640, 5660, 5680, 5700, 5745, 5765, 5785, 5805, 5825, 5845, 5855, 5860, 5865, 5870, 5875, 5880, 5885, 5890, 5895, 5900, 5905, 5910, 5915, 5920, 5925
+
+ieee80211_raw_rates = 10, 20, 55, 60, 90, 110, 120, 180, 240, 360, 480, 540
+
+# Combined set of 802.11 rates for 5Mhz, 10Mhz and other channel widths
+# Only rates for drivers/net/wireless/mac80211_hwsim.c are defined
+ieee80211_rate_label = 1, 2, 3, 4, 5, 6, 9, 11, 12, 18, 22, 24, 27, 36, 48, 54, 72, 96, 108
+
+# Rates as they are represented (see 9.4.2.3 of IEEEE 802.11-2016)
+ieee80211_rate {
+	label		flags[ieee80211_rate_label, int8:7]
+	mandatory	int8:1
+} [packed]
+
+type ieee80211_beacon_interval[BASE_TYPE] [
+	default	const[100, BASE_TYPE]
+	random	BASE_TYPE
+]
+
+type ieee80211_timestamp int64
+
+# Pseudo syscalls and initially created devices use the default frequency below 
+type ieee80211_frequency_mhz[BASE_TYPE] [
+	default	const[2412, BASE_TYPE]
+	random	flags[ieee80211_channel_freq_mhz, BASE_TYPE]
+]
+
+# Information Element structure (see 9.4.2.1 of IEEE 802.11-2016)
+# Also see `element` struct in include/linux/ieee80211.h
+type ieee80211_generic_ie[ID_TYPE, DATA_TYPE] {
+	id	ID_TYPE
+	len	len[data, int8]
+	data	DATA_TYPE
+} [packed]
+
+ieee80211_ie [
+	random	ieee80211_generic_ie[int8, array[int8, 0:255]]
+] [varlen]
+
+# HT Capabilities (see 9.4.2.56 of IEEE 802.11-2016)
+ieee80211_ht_cap_info = IEEE80211_HT_CAP_LDPC_CODING, IEEE80211_HT_CAP_SUP_WIDTH_20_40, IEEE80211_HT_CAP_SM_PS, IEEE80211_HT_CAP_SM_PS_SHIFT, IEEE80211_HT_CAP_GRN_FLD, IEEE80211_HT_CAP_SGI_20, IEEE80211_HT_CAP_SGI_40, IEEE80211_HT_CAP_TX_STBC, IEEE80211_HT_CAP_RX_STBC, IEEE80211_HT_CAP_RX_STBC_SHIFT, IEEE80211_HT_CAP_DELAY_BA, IEEE80211_HT_CAP_MAX_AMSDU, IEEE80211_HT_CAP_DSSSCCK40, IEEE80211_HT_CAP_RESERVED, IEEE80211_HT_CAP_40MHZ_INTOLERANT, IEEE80211_HT_CAP_LSIG_TXOP_PROT
+
+ieee80211_ht_ext_cap_info = IEEE80211_HT_EXT_CAP_PCO, IEEE80211_HT_EXT_CAP_PCO_TIME, IEEE80211_HT_EXT_CAP_PCO_TIME_SHIFT, IEEE80211_HT_EXT_CAP_MCS_FB, IEEE80211_HT_EXT_CAP_MCS_FB_SHIFT, IEEE80211_HT_EXT_CAP_HTC_SUP, IEEE80211_HT_EXT_CAP_RD_RESPONDER
+
+# See 9.4.2.56.4 of IEEE 802.11-2016 and the `ieee80211_mcs_info` struct
+ieee80211_ht_mcs_set {
```

Comment:
Is this the same struct as `ieee80211_mcs_info`? If so let's name it `ieee80211_mcs_info`.